core/code/hooks.js

```diff
@@ -99,11 +102,16 @@ window.removeHook = function(event, callback) {
     throw new Error('Callback must be a function.');
   }
 
-  if (_hooks[event]) {
-    var index = _hooks[event].indexOf(callback);
+  var listeners = _hooks[event];
+  if (listeners) {
+    var index = listeners.indexOf(callback);
     if(index == -1)
       log.warn('Callback wasn\'t registered for this event.');
     else
-      _hooks[event].splice(index, 1);
+      if (isRunning) {
+        listeners[index] = L.Util.falseFn;
+        _hooks[event] = listeners = listeners.slice();
```

Comment:
chain assignment - some may like them, some may not.

same code, same performance:
```suggestion
        listeners = listeners.slice();
        _hooks[event] = listeners;
```

Reply:
> chain assignment - some may like them, some may not.

I have no own opinion about this (but do not use it myself).
But here is just copy of Leaflet code, and I do not see strong reason to alter it.